botocore/handlers.py

```diff
@@ -519,6 +519,20 @@ def change_get_to_post(request, **kwargs):
         request.url, request.data = request.url.split('?', 1)
 
 
+def set_list_objects_encoding_type_url(params, **kwargs):
+    if 'EncodingType' not in params:
+        params['EncodingType'] = 'url'
+
+
+def decode_list_object(parsed, **kwargs):
+    # This is needed because we are passing url as the encoding type. Since the
+    # paginator is based on the key, we need to handle it before it can be
+    # round tripped.
+    if 'Contents' in parsed and parsed.get('EncodingType') == 'url':
```

Comment:
I wonder if we should only automatically decode the URI encoded values if the EncodingType was not explicitly set by the user? For example, if a user actually looks at the S3 API and sees that you can set and EncodingType, then they might reasonably expect that the results they receive are encoded... In fact, that's probably the behavior they are expecting right now. If we automatically start decoding these, it could be a breaking change.